shared/templates/create_permissions.py

```diff
@@ -0,0 +1,217 @@
+                        {
+                            "FILEPATH":      full_path,
+                            "FILEUID":       uid,
+                        },
+                        "./bash/file_owner{0}.sh", path_id
+                    )
+                if gid:
+                    self.file_from_template(
+                        "./template_BASH_file_groupowner",
+                        {
+                            "FILEPATH":      full_path,
+                            "FILEGID":       gid,
+                        },
+                        "./bash/file_groupowner{0}.sh", path_id
+                    )
+                self.file_from_template(
+                    "./template_BASH_permissions",
+                    {
+                        "FILEPATH":      full_path,
+                        "FILEMODE":      mode,
+                        "FILEUID":       uid,
+                        "FILEGID":       gid,
+                    },
+                    "./bash/permissions{0}.sh", path_id
+                )
+            else:
+                file_name = re.sub('^\^', '', file_name)
+                self.file_from_template(
+                    "./template_BASH_file_regex_permissions",
+                    {
+                        "FILEPATH":      dir_path,
+                        "FILENAME":      file_name,
+                        "FILEMODE":      mode,
+                    },
+                    "./bash/file_permissions{0}.sh", path_id
+                )
+
+        elif target == "ansible" and not re.match(r'\^.*\$', file_name, 0):
+            if mode:
+                self.file_from_template(
+                    "./template_ANSIBLE_file_permissions",
+                    {
+                        "FILEPATH":      full_path,
+                        "FILEMODE":      mode,
+                    },
+                    "./ansible/file_permissions{0}.yml", path_id
+                )
+            if uid:
+                self.file_from_template(
+                    "./template_ANSIBLE_file_owner",
+                    {
+                        "FILEPATH":      full_path,
+                        "FILEUID":       uid,
+                    },
+                    "./ansible/file_owner{0}.yml", path_id
+                )
+            if gid:
+                self.file_from_template(
+                    "./template_ANSIBLE_file_groupowner",
+                    {
+                        "FILEPATH":      full_path,
+                        "FILEGID":       gid,
+                    },
+                    "./ansible/file_groupowner{0}.yml", path_id
+                )
+
+            self.file_from_template(
+                "./template_ANSIBLE_permissions",
+                {
+                    "FILEPATH":      full_path,
+                    "FILEMODE":      mode,
+                    "FILEUID":       uid,
+                    "FILEGID":       gid,
+                },
+                "./ansible/permissions{0}.yml", path_id
```

Comment:
Same as with `template_BASH_permission`, generate only if `mode`, uid,` and `gid` are defined.

Reply:
(same as above: see https://github.com/cipherboy/scap-security-guide/blob/banach-space/shared/templates/template_ANSIBLE_permissions)